src/main/java/org/quiltmc/loader/impl/util/ImmutableIterable.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.loader.impl.util;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Iterator;
+
+public final class ImmutableIterable<T> implements Iterable<T> {
```

Comment:
I don't understand how this makes the Iterable immutable? You never clone, just keep a reference to the inital object.

Reply:
`Iterator#remove` throws an `UnsupportedOperationException` by default. By wrapping whatever iterable is passed with a new subclass of `Iterator`, attempts to call `Iterator#remove` will result in throwing that exception.